create_per_gene_matrix.py

```diff
@@ -60,321 +177,319 @@ def get_worst_gene_csq_code_expr(vep_expr: hl.expr.StructExpression) -> hl.expr.
-                (vp_mt.popmax_af <= 0.001),
-                vp_mt.pop_af[vp_mt.pop] <= 0.001
-            )
-            & vp_mt.x,
-            get_grouped_phase_agg()
+        .when(
+            ~vp_mt.is_singleton_vp
+            & (vp_mt.phase_info["all"].em.adj.p_chet <= SAME_HAP_THRESHOLD),
+            "same_hap",
         )
+        .default("unphased"),
     )
 
-    vp_mt = vp_mt.checkpoint('gs://gnomad-tmp/compound_hets/chet_per_gene{}.2.mt'.format(
-        '.chr20' if chr20 else ''
-    ), overwrite=True)
+    logger.info(
+        "Exploding rows by VEP annotation (list of structs with 'gene_id', 'gene_symbol', 'csq' annotations), allele "
+        "frequency cutoffs..."
+    )
+    vp_mt = vp_mt.explode_rows("vep")
+    vp_mt = vp_mt.transmute_rows(**vp_mt.vep)
+    vp_mt = vp_mt.explode_rows(vp_mt.af_cutoff)
+    vp_mt = vp_mt.explode_rows(vp_mt.csq)
+
+    logger.info("Checkpointing exploded variant pair MatrixTable...")
+    vp_mt = vp_mt.checkpoint(
+        f"gs://gnomad-tmp/compound_hets/chet_per_gene.rows_exploded{'.test' if test else ''}.mt",
+        _read_if_exists=not overwrite,
+        overwrite=overwrite,
+    )
 
-    gene_ht = vp_mt.annotate_rows(
-        row_counts=hl.flatten([
-            hl.array(
-                hl.agg.group_by(
-                    vp_mt.pop,
-                    hl.struct(
-                        csq=csq,
-                        af=af,
-                        # TODO: Review this
-                        # These will only kept the worst csq -- now maybe it'd be better to keep either
-                        # - the worst csq for chet or
-                        # - the worst csq for both chet and same_hap
-                        n_worst_chet=hl.agg.count_where(vp_mt[af].get(1) == csq_i),
-                        n_chet=hl.agg.count_where((vp_mt[af].get(1) == csq_i) & (vp_mt[af].get(2, 9) >= csq_i) & (vp_mt[af].get(3, 9) >= csq_i)),
-                        n_same_hap=hl.agg.count_where((vp_mt[af].get(2) == csq_i) & (vp_mt[af].get(1, 9) > csq_i) & (vp_mt[af].get(3, 9) >= csq_i)),
-                        n_unphased=hl.agg.count_where((vp_mt[af].get(3) == csq_i) & (vp_mt[af].get(1, 9) > csq_i) & (vp_mt[af].get(2, 9) > csq_i))
-                    )
-                )
-            ).filter(
-                lambda x: (x[1].n_chet > 0) | (x[1].n_same_hap > 0) | (x[1].n_unphased > 0)
-            ).map(
-                lambda x: x[1].annotate(
-                    pop=x[0]
-                )
+    logger.info(
+        "Aggregating variant pairs by gene_id, gene_symbol, csq, and af_cutoff and for each sample adding entry "
+        "annotations indicating if the sample contains any variant pair (het_het) and annotations for 'chet', "
+        "'same_hap', and 'unphased' if the sample contains any variant pair within those co-occurrence groupings..."
+    )
+    vp_mt = vp_mt.group_rows_by("gene_id", "gene_symbol", "csq", "af_cutoff").aggregate(
```

Comment:
I think we can make this all faster with, it will be the same, but less rows after explode and therefore less rows to aggregate overall. Also make a `PHASE_GROUPS` global variablel
```
    vp_mt = (
        vp_mt.group_rows_by("gene_id", "gene_symbol", "csq", "af_cutoff")
        .aggregate(
            **{
                chet_group: hl.agg.filter(
                    vp_mt.chet_group == chet_group, hl.agg.any(vp_mt.x)
                )
                for chet_group in PHASE_GROUPS
            },
            het_het=hl.agg.any(vp_mt.x),
        )
        .key_rows_by("gene_id", "gene_symbol", "csq")
    )
    vp_mt = vp_mt.explode_rows(vp_mt.af_cutoff)

    vp_mt = vp_mt.group_rows_by("gene_id", "gene_symbol", "csq", "af_cutoff").aggregate(
        **{chet_group: hl.agg.any(vp_mt[chet_group]) for chet_group in PHASE_GROUPS},
        het_het=hl.agg.any(vp_mt.het_het),
    )
```